dora/core/client/fs/src/main/java/alluxio/client/file/DoraFileOutStream.java

```diff
@@ -0,0 +1,256 @@
+    mDoraClient = doraClient;
+    mNettyDataWriter = dataWriter;
+    mCloser = Closer.create();
+    mUuid = uuid;
+    // Acquire a resource to block FileSystemContext reinitialization, this needs to be done before
+    // using mContext.
+    // The resource will be released in close().
+    mContext = context;
+    mCloser.register(mContext.blockReinit());
+    try {
+      mUri = Preconditions.checkNotNull(path, "path");
+      mAlluxioStorageType = options.getAlluxioStorageType();
+      mUnderStorageType = options.getUnderStorageType();
+      mOptions = options;
+      mClosed = false;
+      mCanceled = false;
+      mWriteToAlluxio = mAlluxioStorageType.isStore();
+      mBytesWritten = 0;
+
+      if (!mUnderStorageType.isSyncPersist()) {
+        mUnderStorageOutputStream = null;
+      } else { // Write is through to the under storage, create mUnderStorageOutputStream.
+        mUnderStorageOutputStream = ufsOutStream;
+      }
+    } catch (Throwable t) {
+      throw CommonUtils.closeAndRethrow(mCloser, t);
+    }
+  }
+
+  @Override
+  public void cancel() throws IOException {
+    mCanceled = true;
+    close();
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (mClosed) {
+      return;
+    }
+    try (Timer.Context ctx = MetricsSystem
+            .uniformTimer(MetricKey.CLOSE_ALLUXIO_OUTSTREAM_LATENCY.getName()).time()) {
+      mNettyDataWriter.flush();
+      mNettyDataWriter.close();
+
+      CompleteFilePOptions.Builder optionsBuilder = CompleteFilePOptions.newBuilder();
+      optionsBuilder.setCommonOptions(FileSystemMasterCommonPOptions.newBuilder()
+          .setOperationId(new OperationId(UUID.randomUUID()).toFsProto()).buildPartial());
+      if (mUnderStorageType.isSyncPersist()) {
+        if (mCanceled) {
+          mUnderStorageOutputStream.cancel();
+        } else {
+          mUnderStorageOutputStream.flush();
+          mUnderStorageOutputStream.close();
+          optionsBuilder.setUfsLength(mBytesWritten);
+        }
+      }
+
+      if (mAlluxioStorageType.isStore()) {
+        if (mCanceled) {
+          mNettyDataWriter.cancel();
+        } else {
+          // Note, this is a workaround to prevent commit(blockN-1) and write(blockN)
+          // race, in worse case, this may result in commit(blockN-1) completes earlier than
+          // write(blockN), and blockN evicts the committed blockN-1 and causing file lost.
+        }
+      }
+
+      // Whether to complete file with async persist request.
+      if (!mCanceled && mUnderStorageType.isAsyncPersist()
+          && mOptions.getPersistenceWaitTime() != Constants.NO_AUTO_PERSIST) {
+        optionsBuilder.setAsyncPersistOptions(
+            FileSystemOptionsUtils.scheduleAsyncPersistDefaults(
+                mContext.getPathConf(mUri)).toBuilder()
+                .setCommonOptions(mOptions.getCommonOptions())
+                .setPersistenceWaitTime(mOptions.getPersistenceWaitTime()));
+      }
```

Comment:
If we don't support async writes right now, let's remove the code to avoid confusion.